### DISCUSSION TITLE
Feature request: TimeDistributed Layer

### DISCUSSION THREAD

**Discussion by NeroBlackstone** [LuxDL/Lux.jl#773] [Q&A] [answered]
Motivation:
If [Lux.Recurrence ](https://lux.csail.mit.edu/stable/api/Lux/layers#Linear-Layers) `return_sequence` arg is `true`, it will return a vector of 2d-matrix.

``` julia
model = Chain(Recurrence(GRUCell(3=>32),return_sequence = true))

rng = Random.default_rng()
Random.seed!(rng, 0)
ps, st = Lux.setup(rng, model)

x = rand(Float32,3,4,5)
y1 = model(x,ps,st)[1]

@show size(y1)
@show size(y1[1])

# sequence_length
#size(y1) = (4,)
# output_size * batch_size
#size(y1[1]) = (32, 5)
```

Suppose we have a sequence labeling task, and the output of each time step is fed into a Dense layer (as a classifier) ​​for classification, looks like this:

![TimeDistributed](https://github.com/LuxDL/Lux.jl/assets/17470335/8d7a4a32-850a-4e12-bef5-f88abc797adb)

Can the existing layers do this? What do you think?

I think it works like this:

``` julia
model = Chain(Recurrence(GRUCell(3=>32),return_sequence = true),TimeDistributed(Dense(32=>2)))
rng = Random.default_rng()
Random.seed!(rng, 0)
ps, st = Lux.setup(rng, model)

x = rand(Float32,3,4,5)
y1 = model(x,ps,st)[1]

@show size(y1)
@show size(y1[1])

# sequence_length
#size(y1) = (4,)
# output_size * batch_size
#size(y1[1]) = (2, 5)
```

I have not studied [TimeDistributed](https://keras.io/api/layers/recurrent_layers/time_distributed/) in Keras, and I don’t know if my understanding is correct.

Is it necessary to have a function as a parameter in order to merge the output of TimeDistributed? like:

``` julia
...
TimeDistributed(Dense(32=>2);vcat)
...
# output
# sequence_length (after vcat) * batch_size
#size(y1) = (8, 5)
```

Do you think this feature is necessary? If so, I can implement it. Please tell me some ideas (such as existing layers that can be referenced) and I will complete it. Thank you!

**Comment by avik-pal** [comment, marked as answer]
This is a special case of https://github.com/LuxDL/Lux.jl/issues/426 where the number of layers is `1`.
